clients/gitlabrepo/commits.go

```diff
@@ -27,31 +27,52 @@ import (
 )
 
 type commitsHandler struct {
-	glClient   *gitlab.Client
-	once       *sync.Once
-	errSetup   error
-	repourl    *repoURL
-	commitsRaw []*gitlab.Commit
+	glClient    *gitlab.Client
+	once        *sync.Once
+	errSetup    error
+	repourl     *repoURL
+	commitsRaw  []*gitlab.Commit
+	commitDepth int
 }
 
-func (handler *commitsHandler) init(repourl *repoURL) {
+func (handler *commitsHandler) init(repourl *repoURL, commitDepth int) {
 	handler.repourl = repourl
 	handler.errSetup = nil
 	handler.once = new(sync.Once)
+	handler.commitDepth = commitDepth
 }
 
 func (handler *commitsHandler) setup() error {
 	handler.once.Do(func() {
-		commits, _, err := handler.glClient.Commits.ListCommits(
-			handler.repourl.projectID,
-			&gitlab.ListCommitsOptions{
-				RefName: &handler.repourl.commitSHA,
-			},
-		)
-		if err != nil {
-			handler.errSetup = fmt.Errorf("request for commits failed with %w", err)
-			return
+		var commits []*gitlab.Commit
+		i := 1
+
+		for {
+			c, _, err := handler.glClient.Commits.ListCommits(handler.repourl.projectID,
+				&gitlab.ListCommitsOptions{
+					ListOptions: gitlab.ListOptions{
+						Page:    i,
+						PerPage: handler.commitDepth,
```

Comment:
Are there any limits on the GitLab side of things for how many commits we can ask for? On the GitHub side we have to request max of 100 at a time for example.

Reply:
So the Gitlab default for PerPage is 20, max of 100. However, if you enter a commit depth of 200 gitlab will still return a perpage of 100 and we just iterate until we hit the requested depth

Reply:
Got it, do you know if it's the `Page` value depends on `PerPage` at all? For example let's say we request a commit depth of 200. This is too large for the first iteration (gitlab only returns commits 1-100 as you said). On the second iteration we'd be sending something like:
```
Page: 2
PerPage: 200
```


Again, this will return 100 commits, but does it return commits 101 - 200? or commits 201-300?

Reply:
It would return commits 101-200, we're just getting the second page (of 100) of repository commits. I'm trying to find a good example in the docs, but nothing really illustrates this. Theres [this](https://github.com/xanzy/go-gitlab/blob/793bc3cdf558e2ec74e5873e1de9fcd535266aa0/examples/pagination.go#L25) from the go-gitlab docs which is essentially what we're doing.

Reply:
Since this is an example of what we're doing, can we also handle pagination the same way, using the semantics from the example? I think the readability is better

```go
		// Exit the loop when we've seen all pages.
		if resp.NextPage == 0 {
			break
		}

		// Update the page number to get the next page.
		opt.Page = resp.NextPage
```

Reply:
Yeah, I think thats a good idea